modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/sequenceK.kt

```diff
@@ -372,3 +376,33 @@ interface SequenceKUnzip : Unzip<ForSequenceK>, SequenceKZip {
       (seq.map { it.a }.k() toT seq.map { it.b }.k())
     }
 }
+
+@extension
+interface SequenceKCrosswalk : Crosswalk<ForSequenceK>, SequenceKFunctor, SequenceKFoldable {
+  override fun <F, A, B> crosswalk(ALIGN: Align<F>, a: Kind<ForSequenceK, A>, fa: (A) -> Kind<F, B>): Kind<F, Kind<ForSequenceK, B>> {
+    fun crosswalkSequence(iterator: Iterator<A>): Kind<F, Kind<ForSequenceK, B>> =
+      if (iterator.hasNext()) {
+        val head = iterator.next()
+        val ls = fa(head)
+        val rs = crosswalkSequence(iterator)
+
+        ALIGN.run {
+          alignWith(ls, rs) { ior ->
+            ior.fold({ SequenceK.just(it) },
+              ::identity,
+              { l, r -> (SequenceK.just(l) + r.fix().sequence).k() })
+          }
+        }
+      } else {
+        ALIGN.run { empty<B>().map { SequenceK.empty<B>() } }
+      }
+
+    return crosswalkSequence(a.fix().sequence.iterator())
+  }
```

Comment:
Same as with the list. This cannot be fully lazy anyway so it might as well use a fold.

Reply:
ok, it uses the list implementation now